controllers/operator/suite_test.go

```diff
@@ -112,6 +132,62 @@ func startTestEnv() {
 	Expect(err).NotTo(HaveOccurred())
 	Expect(k8sClient).NotTo(BeNil())
 	//+kubebuilder:scaffold:scheme
+
+	// Step 9: Apply the resources from the buffer to the test environment
+	klog.Info("Applying paches to the test environment")
+	err = applyResources(buffy)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func applyResources(buffer *bytes.Buffer) error {
+	// Create a universal decoder for deserializing the resources
+	decoder := scheme.Codecs.UniversalDeserializer()
+
+	// Split the buffer into individual YAML documents
+	docs := bytes.Split(buffer.Bytes(), []byte("\n---[ ]*\n"))
+	for _, doc := range docs {
```

Comment:
```suggestion
	klog.Info("Applying CRDs to the test environment")
	kustomizer := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
	resMap, err := kustomizer.Run(filesys.MakeFsOnDisk(), filepath.Join("..", "..", "config", "crd"))
	Expect(err).NotTo(HaveOccurred())
	err = applyResources(resMap)
	Expect(err).NotTo(HaveOccurred())
}

func applyResources(resources resmap.ResMap) error {
	// Create a universal decoder for deserializing the resources
	decoder := scheme.Codecs.UniversalDeserializer()

	for _, res := range resources.Resources() {
		raw, err := res.AsYAML()
		Expect(err).NotTo(HaveOccurred())
```

We can build the kustomize-d resources and directly use the returned `resMap.Resources()` as a list of documents to apply